packages/backend/src/services/RenameService/rename.ts

```diff
@@ -0,0 +1,823 @@
+        replaceFieldReference = (str: string) => str; // no changes
+        replaceFieldName = (str: string) => str; // no changes
+    } else {
+        replaceFieldName = (str: string) =>
+            str.replace(
+                new RegExp(`^${fromFieldName}$`, 'g'),
+                `${toFieldName}`,
+            ); // field name (eg: customer_id)
+        replaceId = (str: string) =>
+            str.replace(new RegExp(`^${from}$`, 'g'), `${to}`); // entire id (eg: payment_customer_id)
+        replaceReference = (str: string) =>
+            str.replace(
+                new RegExp(`\\$\\{${fromReference}\\}`, 'g'),
+                `\${${toReference}}`,
+            ); // Full reference
+        replaceString = (str: string) =>
+            str.replace(new RegExp(`${from}`, 'g'), `${to}`);
+        replaceFieldReference = (str: string) => {
+            /// These are for ${TABLE}.field references like in additionalMetrics
+            if (str.startsWith('${TABLE}')) {
+                const fromReferenceWithoutTable = fromReference.split('.')[1];
+                const toReferenceWithoutTable = toReference.split('.')[1];
+                return str.replace(
+                    fromReferenceWithoutTable,
+                    toReferenceWithoutTable,
+                );
+            }
+            return replaceReference(str);
+        };
+    }
+
+    return {
+        replaceId,
+        replaceString,
+        replaceReference,
+        replaceOptionalId: (str: string | undefined) =>
+            str ? replaceId(str) : undefined,
+        replaceKeys: <T>(obj: Record<string, T>) => {
+            const keyValues = Object.keys(obj).map((key) => {
+                const newKey = replaceId(key);
+                return { [newKey]: obj[key] };
+            });
+            return Object.assign({}, ...keyValues);
+        },
+        replaceFull: (str: string) => str.replaceAll(from, to), // Full name replace, used in tables
+        replaceList: (list: string[]) => list.map((item) => replaceId(item)),
+        replaceOptionalList: (list: string[] | undefined) =>
+            list ? list.map((item) => replaceId(item)) : undefined,
+        replaceFieldReference,
+        replaceFieldName,
+    };
+};
+
+/*
+We are renaming fields in the object 
+This is only for validating that the we are replacing all the fields needed
+*/
+export const validateRename = (
+    originalObject: Object,
+    updatedObject: Object,
+    objectName: string,
+    { from, fromReference, to, toReference }: NameChanges,
+) => {
+    try {
+        // Instead of trying to compare the entire objects, let's just compare the stringified versions
+        // after normalizing them (parsing and stringifying to remove any non-enumerable properties)
+        const normalizeAndReplace = (obj: AnyType): string =>
+            JSON.stringify(JSON.parse(JSON.stringify(obj)))
+                .replaceAll(from, to)
```

Comment:
to avoid unnecessary replaces, and because both of them are strings, I think you can add `"${from}"` and `"${to}"`